### DISCUSSION TITLE
Hide tray icon

### DISCUSSION THREAD

**Discussion by Razer0123** [streamdeck-linux-gui/streamdeck-linux-gui#120] [Ideas / Feature Requests]
I just use this app to disable the Streamdeck on Linux, is it possible to have an option to disable the tray icon?

**Comment by Danie10** [comment]
@Razer0123 that option is already supported by the OS itself - well on Plasma anyway. By going to configure system tray settings, you can choose between "shown when relevant", "always shown", "always hidden".

![Screenshot_20231122_195747](https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/assets/1153726/81314bba-bfef-426b-9a8c-40c14666ce31)

**Comment by coolapso** [comment]
Don't think this is a feature that should be provided by the application.  
* Tray icons behave differently depending on the desktop environment / compositor / Graphical environment. 
* Tray Icons is a feature from the Desktop environments and enabling / disabling it all together its a feature provided by such as well as enabling/disabling specific ones just like @Danie10 demonstrated.

**Reply by Razer0123** [reply]
I am on Gnome and i have no option to disable Streamdeck tray icon, otherwise i would not have posted

Also several apps include an option to disable tray icons

**Comment by Danie10** [comment]
OK I don't use GNOME so can't test this but according to Google Bard there may be some ways to do it:

Disabling system tray icons in GNOME Linux can be achieved through various methods depending on your specific GNOME version and preferences. Here are two common approaches:

**Method 1: Using GNOME Extensions**

GNOME Extensions provide a convenient way to customize the GNOME desktop, including the ability to disable system tray icons. To use this method, follow these steps:

Install GNOME Extensions: Ensure GNOME Extensions is installed on your system. If not, you can install it from the GNOME Software Center or by running the following command in the terminal:
`sudo apt install gnome-shell-extensions`
Install the "Hide Applications in System Tray" extension: Open the GNOME Extensions website and search for the "Hide Applications in System Tray" extension. Click the "Install" button and follow the prompts to install the extension.

Enable the extension: Once installed, open the "Extensions" application from the GNOME Activities panel. Find the "Hide Applications in System Tray" extension and toggle the switch to enable it.

Configure the extension: Right-click on the extension icon in the GNOME Activities panel and select "Configure." Choose the desired options to customize how system tray icons are hidden.

**Method 2: Using dconf-editor**

dconf-editor is a system configuration tool that allows you to modify various GNOME settings, including those related to system tray icons. To use this method, follow these steps:

Install dconf-editor: If not already installed, install dconf-editor using the following command in the terminal:
`sudo apt install dconf-editor`
Launch dconf-editor: Open the dconf-editor application.

Navigate to the relevant settings: Navigate to the following path in dconf-editor:

`/org/gnome/desktop/interface/icon-theme`
Disable system tray icons: Locate the "use-system-tray" setting and change its value to "false." This will disable system tray icons for all applications.

Restart GNOME Shell: For the changes to take effect, restart GNOME Shell using the following command in the terminal:

`Alt+F2 > r > Enter`

These methods should effectively disable system tray icons in GNOME Linux. If you encounter any issues or prefer a different approach, feel free to search for alternative methods or consult the GNOME documentation for further assistance.

**Comment by Razer0123** [comment]
> OK I don't use GNOME so can't test this but according to Google Bard there may be some ways to do it:
> 
> Disabling system tray icons in GNOME Linux can be achieved through various methods depending on your specific GNOME version and preferences. Here are two common approaches:
> 
> **Method 1: Using GNOME Extensions**
> 
> GNOME Extensions provide a convenient way to customize the GNOME desktop, including the ability to disable system tray icons. To use this method, follow these steps:
> 
> Install GNOME Extensions: Ensure GNOME Extensions is installed on your system. If not, you can install it from the GNOME Software Center or by running the following command in the terminal: `sudo apt install gnome-shell-extensions` Install the "Hide Applications in System Tray" extension: Open the GNOME Extensions website and search for the "Hide Applications in System Tray" extension. Click the "Install" button and follow the prompts to install the extension.
> 
> Enable the extension: Once installed, open the "Extensions" application from the GNOME Activities panel. Find the "Hide Applications in System Tray" extension and toggle the switch to enable it.
> 
> Configure the extension: Right-click on the extension icon in the GNOME Activities panel and select "Configure." Choose the desired options to customize how system tray icons are hidden.
> 
> **Method 2: Using dconf-editor**
> 
> dconf-editor is a system configuration tool that allows you to modify various GNOME settings, including those related to system tray icons. To use this method, follow these steps:
> 
> Install dconf-editor: If not already installed, install dconf-editor using the following command in the terminal: `sudo apt install dconf-editor` Launch dconf-editor: Open the dconf-editor application.
> 
> Navigate to the relevant settings: Navigate to the following path in dconf-editor:
> 
> `/org/gnome/desktop/interface/icon-theme` Disable system tray icons: Locate the "use-system-tray" setting and change its value to "false." This will disable system tray icons for all applications.
> 
> Restart GNOME Shell: For the changes to take effect, restart GNOME Shell using the following command in the terminal:
> 
> `Alt+F2 > r > Enter`
> 
> These methods should effectively disable system tray icons in GNOME Linux. If you encounter any issues or prefer a different approach, feel free to search for alternative methods or consult the GNOME documentation for further assistance.

Appreciate the help, but this would disable ALL tray icons, i would like to only disable this one

I know it's not a major issue, but was worth asking
